features/0454-present-proof-v2/README.md

```diff
@@ -0,0 +1,347 @@
+* `request-presentation` - Verifier to Prover - request a presentation
+* `presentation` - Prover to Verifier - provide a presentation in response to a request
+
+In addition, the [`ack`](../0015-acks/README.md) and [`problem-report`](../0035-report-problem/README.md) messages are adopted into the protocol for confirmation and error handling.
+
+The messages that include `~attach` attachments may use either form of the embedded attachment&mdash;`base64` or `json`. In the examples below, one or the other is arbitrarily used.
+
+The `~attach` array is to be used to enable a single presentation to be requested/delivered in different verifiable presentation formats. The ability to have multiple attachments must not be used to request/deliver multiple different presentations in a single instance of the protocol.
+
+### Propose Presentation
+
+An optional message sent by the prover to the verifier to initiate a proof presentation process, or in response to a `request-presentation` message when the prover wants to propose using a different presentation format or request. Schema:
+
+```json
+{
+    "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/present-proof/%VER/propose-presentation",
+    "@id": "<uuid-propose-presentation>",
+    "comment": "some comment",
+    "formats" : [
+        {
+            "attach_id" : "<attach@id value>",
+            "format" : "<format-and-version>",
+        }
+    ],
+    "proposal~attach": [
+        {
+            "@id": "<attachment identifier>",
+            "mime-type": "application/json",
+            "data": {
+                "json": "<bytes for json>"
+            }
+        }
+    ]
+}
+```
+
+Description of fields:
+
+* `comment` -- a field that provides some human readable information about the proposed presentation.
+* `formats` -- contains an entry for each `filter~attach` array entry, including an optional value of the attachment `@id` (if attachments are present) and the verifiable presentation format and version of the attachment. Accepted values for the `format` items are provided in the per format "Attachment" sections immediately below.
+* `proposal~attach` -- an optional array of attachments that further define the presentation request being proposed. This might be used to clarify which formats or format versions are wanted.
+
+If the `proposal~attach` is not provided, the `attach_id` item in the `formats` array should not be provided. That form of the `propose-presentation` message is to indicate the presentation formats supported by the prover, independent of the verifiable presentation request content.
+
+#### Negotiation and Preview
+
+Negotiation prior to the delivery of the presentation can be done using the `propose-presentation` and `request-presentation` messages. The common negotiation use cases would be about the claims to go into the presentation and the format of the verifiable presentation.
+
+#### Propose Attachment Registry
+
+##### Hyperledger Indy Proposal Attachment
```

Comment:
V1 of this protocol has been criticized as being Indy-centric. We know that was not the intent, but that's the way it reads to a non-Indy audience. It would be nice to include some a section for something besides Indy. If we can't add such a section because we don't know what it would contain, can we create a placeholder with "TODO" text? And can we put it before Indy?

Reply:
This will be added shortly (follow on PR) to include the DIF presentation definition.  Leaving off for now.